staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -257,10 +276,76 @@ func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta
 		return false, apierrors.NewInternalError(err)
 	}
 
-	changed := !apiequality.Semantic.DeepEqual(attr.VersionedObject, newVersionedObject)
+	changed = !apiequality.Semantic.DeepEqual(attr.VersionedObject, newVersionedObject)
 	trace.Step("Patch applied")
+	if changed {
+		annotator.addPatchAnnotation(result.Patch, result.PatchType)
```

Comment:
can we pass `patchObj` instead to save decoding the patch, and make `addPatchAnnotation` take a `patch interface{}` parameter, with the requirement that the parameter be json-encodeable?

Reply:
switched to passing `patchObj` as an `interface{}` and encoding the Go type to JSON

an additional change: log the patch regardless of `changed`